test/distributions/block/sliceComm/NUMLOCALES

```diff
@@ -0,0 +1 @@
+16
```

Comment:
Is there a big advantage to using 16 locales instead of say 4? 16 locales when oversubscribed will really bog down our less beefy systems (darwin testing, or probably any non-chapcs testing.)

Reply:
For 4 locales, the comm reductions for the sliceComm ones aren't as interesting or dramatic since sqrt(4) = 4/2 = 2 (i.e., half the locales are still involved even if you optimize away the other half; and for the inner slice, all the locales are still involved).  It didn't seem to bog down my Mac too much, but I could definitely dial it down if you thought it was worth it, particularly for the non-sliceComm tests.  Do we do oversubscription with gasnet testing that will make things worse than what I saw on my Mac?  (realizing that not all our test machines are as awesome as my Mac).

Reply:
I don't think we have any gasnet testing that runs on anything less powerful than chapelmac.

Here's example timings I see on chapelmac though (timings are similar on my mac): 

```
$ time -p ./sliceAssigns -nl 4 >/dev/null
real 0.74
user 0.00
sys 0.00
$ time -p ./sliceAssigns -nl 9 >/dev/null
real 4.87
user 0.00
sys 0.00
$ time -p ./sliceAssigns -nl 16 >/dev/null
real 25.81
user 0.00
sys 0.01
```

~1s for 4 locales, ~5s for 9 locales, and ~26s for 16 locales

26 seconds for a correctness test seems like a lot, but if it makes sense to stick with 16 locales I'm on board with that.